src/cmd/run.go

```diff
@@ -0,0 +1,335 @@
+			} else if containersCount == 1 {
+				containerName = fmt.Sprint(containers[0]["Names"])
+				logrus.Infof("Entering container %s instead", containerName)
+			} else {
+				logrus.Fatal("Create a new container with 'toolbox create' or 'toolbox enter' an already existing container")
+			}
+		} else {
+			logrus.Fatalf("Container '%s' not found", containerName)
+		}
+	}
+
+	// Prepare Flatpak session-helper
+	conn, err := dbus.SessionBus()
+	if err != nil {
+		logrus.Error("Failed to connect to Session Bus")
+	}
+
+	logrus.Info("Calling org.freedesktop.Flatpak.SessionHelper.RequestSession")
+	SessionHelper := conn.Object("org.freedesktop.Flatpak", "/org/freedesktop/Flatpak/SessionHelper")
+	call := SessionHelper.Call("org.freedesktop.Flatpak.SessionHelper.RequestSession", 0)
+	if call.Err != nil {
+		logrus.Debug(call.Err)
+		logrus.Fatal("Failed to call org.freedesktop.Flatpak.SessionHelper.RequestSession")
+	}
+
+	logrus.Infof("Starting container '%s'", containerName)
+	err = containerStart(containerName)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+
+	logrus.Infof("Inspecting container '%s'", containerName)
+	containerInfo, err := podman.PodmanInspect("container", containerName)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+
+	containerConfig := containerInfo["Config"].(map[string]interface{})
+	containerState := containerInfo["State"].(map[string]interface{})
+	containerEntryPoint := containerConfig["Cmd"].([]interface{})[0]
+
+	logrus.Infof("Entry point of container '%s' is '%s'", containerName, containerEntryPoint)
+
+	if containerEntryPoint == "toolbox" {
+		logrus.Infof("Waiting for container '%s' to finish initializing", containerName)
+
+		entryPointPIDString := fmt.Sprint(containerState["Pid"])
+		entryPointPID, err := strconv.Atoi(entryPointPIDString)
+		if err != nil {
+			logrus.Debugf("PID: %s", entryPointPIDString)
+			logrus.Fatalf("Failed to parse entry point PID of container '%s'", containerName)
+		}
+
+		if entryPointPID <= 0 {
+			logrus.Debugf("PID: %d", entryPointPID)
+			logrus.Fatalf("Invalid entry point PID of container '%s'", containerName)
+		}
+
+		// Wait for initialization of a container
+		containerInitializedStamp := fmt.Sprintf("%s/container-initialized-%d", viper.GetString("TOOLBOX_RUNTIME_DIRECTORY"), entryPointPID)
+		logrus.Infof("Checking if initialization stamp %s exists", containerInitializedStamp)
+		containerInitializedTimeout := 25
+		i := 0
+		for !utils.PathExists(containerInitializedStamp) {
+			time.Sleep(time.Second)
+			i++
+
+			if i == containerInitializedTimeout {
+				logrus.Fatalf("Failed to initialize container '%s'", containerName)
+			}
+		}
+		logrus.Infof("Container '%s' is properly initialized", containerName)
+	} else {
+		logrus.Warnf("Container '%s' uses deprecated features", containerName)
+		logrus.Warn("Consider recreating it with Toolbox version 0.0.17 or newer")
+	}
+
+	logrus.Info("Checking if the current working directory is in the container")
+	args = []string{"exec",
```

Comment:
Nitpick: it would be nice to not repurpose the function's `args` parameter for some internal use. It changes the semantics of the variable, which would be fine if it was some loop counter or such, but an incoming argument is a bit more important. :)